src/bindings/rust/README.md

```diff
@@ -27,11 +42,34 @@ fn main() {
 }
 ```
 
+### Key
+
+An example for using a key. For a full example, see the [examples](elektra/src/examples.rs).
+
+```rust
+extern crate elektra;
+use elektra::{KeyBuilder, ReadableKey, StringKey, WriteableKey};
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // To create a simple key with a name and value
+    let mut key = StringKey::new("user/test/language")?;
+    key.set_value("rust");
+
+    assert_eq!(key.name(), "user/test/language");
+    assert_eq!(key.value(), "rust");
+
+    Ok(())
+}
+```
+
+Compared to the C-API, there are two distinct key types, `StringKey` and `BinaryKey`. With these, type mismatches such as calling `keyString` on a `BinaryKey` is not possible. The only difference between them is the type of value you can set and get from them. They are only wrappers over the `Key` from the C-API.
+
+The functionality of the keys is split into two traits, `ReadableKey` and `WritableKey`, which define methods that only read information from a key, and modify a key, respectively. For example, the method to retrieve metakeys only returns a key that implements `ReadableKey`, which is named `ReadOnly`. The keys returned cannot be modified in accordance to the design.
```

Comment:
I did not read anything like that in the api documentation. Where do these occur?

Reply:
For a key that was appended to a KeySet you cannot change the name. I fixed the documentation 23a2f8f756a8c0fdbd980284b7a5ae4ca0802aa2

Reply:
Ok, I was confused because you said there are keys where you can only write to the value, but you probably meant reading?

The `set_name` method handles this by returning an error if the Name is read only (unfortunately this error is called KeyNameInvalidError, because that error can also occur, but both have the same error code).
For simplicity of the design, I decided not to prevent setting the name of a key in a keyset at the type system level.

Reply:
Ok, sorry for the confusion. Please also add this to the docu of set_name.